src/sagas/searchGifs.js

```diff
@@ -0,0 +1,28 @@
+import { takeEvery, call, put } from "redux-saga/effects";
+import axios from "axios";
+
+const fetchGifs = query => {
+  return axios
+    .get("http://api.giphy.com/v1/gifs/search", {
+      params: {
+        q: query,
+        api_key: "NYBl3G1fuM3PcJWfeAv0wSS6fuHAJhIK"
```

Comment:
Use .env variables to save your api keys and do not expose them versioning them. You can provide an .env.example file expressing its usage.

https://www.npmjs.com/package/dotenv